tensorflow/python/keras/metrics.py

```diff
@@ -870,6 +870,30 @@ class SparseCategoricalAccuracy(MeanMetricWrapper):
   def __init__(self, name='sparse_categorical_accuracy', dtype=None):
     super(SparseCategoricalAccuracy, self).__init__(
         sparse_categorical_accuracy, name, dtype=dtype)
+  
+  def update_state(self, y_true, y_pred, sample_weight=None):
+    """Accumulates metric statistics.
```

Comment:
In fact, this docstring update should be applied to some other metric class as well. Can we take this string out as a private const string on the top, and assign it to the __docstring__ field to all the classes that need it?

eg:

```
_UPDATE_STATE_DOCSTRING = """ ... """"

SparseCategoricalAccuracy.update_state.__docstring__ = _UPDATE_STATE_DOCSTRING

```

Reply:
Actually, the docstring for class `SparseCategoricalAccuracy` added in this PR is copied from the base class `MeanMetricWrapper`, with modification on the dimension description. 
https://github.com/tensorflow/tensorflow/blob/8d83bffd4776bbfad12a4e2669e8e30f7a26d3a7/tensorflow/python/keras/metrics.py#L825

The base class `MeanMetricWrapper` docstring:
https://github.com/tensorflow/tensorflow/blob/8d83bffd4776bbfad12a4e2669e8e30f7a26d3a7/tensorflow/python/keras/metrics.py#L628-L644

The child class `SparseCategoricalAccuracy`  docstring:
https://github.com/tensorflow/tensorflow/blob/8d83bffd4776bbfad12a4e2669e8e30f7a26d3a7/tensorflow/python/keras/metrics.py#L874-L890

It is a good idea to take the docstring out to be reused for other `update_state` functions. For example, it seems all Sparse Categorical-based metrics may suffer from this dimension problem and can reuse this docstring. 

On the implementation, taking a doctring to the top and assign it to the `update_state` functions will definitely works. I was wondering if we can use inheritance here as well, adding a pure wrapper, such as

```
class MeanMetricWrapper:
    def update_state(self, y_true, y_pred, sample_weight=None):
        """DocString Version Default
        """
        print('Done MeanMetricWrapper update state')


class SparseCategoricalMetricWrapper(MeanMetricWrapper):
    def update_state(self, y_true, y_pred, sample_weight=None):
        """DocString Version SparseCategorical
        """
        return super(SparseCategoricalMetricWrapper, self).update_state(y_true, y_pred, sample_weight)


class SparseCategoricalAccuracy(SparseCategoricalMetricWrapper):
    pass


if __name__ == '__main__':
    metric = SparseCategoricalAccuracy()
    print(metric.update_state.__doc__)
    metric.update_state(None, None)

>>> 
DocString Version SparseCategorical
        
Done MeanMetricWrapper update state
```

BTW, maybe a little bit off topic, it may be better to take the docstring of the base class out as well. Because the docstring should be highly dependent on the `_fn` member function upon inheritance of the base class `MeanMetricWrapper`. However, it does seem that for now all child metrics of `MeanMetricWrapper`, excluding the sparse categorical ones, require same ranks in dimensions. So practically, we can leave the `MeanMetricWrapper` code as is for the moment. 
https://github.com/tensorflow/tensorflow/blob/8d83bffd4776bbfad12a4e2669e8e30f7a26d3a7/tensorflow/python/keras/metrics.py#L623-L626
https://github.com/tensorflow/tensorflow/blob/8d83bffd4776bbfad12a4e2669e8e30f7a26d3a7/tensorflow/python/keras/metrics.py#L657-L660

Reply:
Just added a wrapper class, and extended the class to SparseCategoricalAccuracy, SparseTopKCategoricalAccuracy and SparseCategoricalCrossentropy.